awx/main/dispatch/pool.py

```diff
@@ -365,6 +367,28 @@ def cleanup(self):
                 logger.warn('scaling down worker pid:{}'.format(w.pid))
                 w.quit()
                 self.workers.remove(w)
+            if w.alive:
+                # if we discover a task manager invocation that's been running
+                # too long, reap it (because otherwise it'll just hold the postgres
+                # advisory lock forever); the goal of this code is to discover
+                # deadlocks or other serious issues in the task manager that cause
+                # the task manager to never do more work
+                current_task = w.current_task
+                if current_task:
+                    if current_task.get('task', '').endswith('tasks.run_task_manager'):
+                        if 'started' not in current_task:
+                            w.managed_tasks[
+                                current_task['uuid']
+                            ]['started'] = datetime.datetime.utcnow()
+                        age = (
+                            datetime.datetime.utcnow() - current_task['started']
```

Comment:
seems like `time.time` would also be fine. Is it more trustworthy than datetimes??

Reply:
Yea, that's a good point, I should just do that.